src/mongo_client.ts

```diff
@@ -118,6 +118,8 @@ export type SupportedNodeConnectionOptions = SupportedTLSConnectionOptions &
 export interface MongoClientOptions extends BSONSerializeOptions, SupportedNodeConnectionOptions {
   /** Specifies the name of the replica set, if the mongod is a member of a replica set. */
   replicaSet?: string;
+  /** @experimental */
+  timeoutMS?: number;
```

Comment:
We could leave this marked internal until it is ready, the tests should rely on the driver `src` type definitions rather than the public-only generated one. 

But this is fine by me, I would add "this option is in development and currently has undefined behavior" or something, because the driver may have partial support as we work to implement it. Thoughts?

Reply:
I've updated to make internal and added a comment.